packages/flutter/lib/src/gestures/events.dart

```diff
@@ -232,6 +233,12 @@ abstract class PointerEvent with Diagnosticable {
   }) : localPosition = localPosition ?? position,
        localDelta = localDelta ?? delta;
 
+  /// Unique identifier that ties the [PointerEvent] to embedder event created it.
+  ///
+  /// No two pointer events can have the same [embedderId]. This is different from
+  /// [pointer] identifier - used for hit-testing, whereas [embedderId] is used to
+  /// identify the platform event.
+  final int embedderId;
```

Comment:
Maybe add a section about Android explaining that on Android this is the ID of the underlying MotionEvent.